typed-function.js

```diff
@@ -91,61 +99,128 @@
 
     var anyType = {
       name: 'any',
-      test: ok
+      test: ok,
+      isAny: true
     }
 
+    // Data structures to track the types. As these are local variables in
+    // create(), each typed universe will get its own copy, but the variables
+    // will only be accessible through the (closures of the) functions supplied
+    // as properties of the typed object, not directly.
+    // These will be initialized in clear() below
+    var typeMap; // primary store of all types
+    var typeList; // Array of just type names, for the sake of ordering
```

Comment:
I think it's a wash, but if you prefer it this way, I'm happy.

Reply:
😂  yes I understand what you mean. Now you have two objects holding the same information that need to be kept in sync with each other. A single LinkedMap would be more neat, having a single source of truth. What I do like about the current approach is that it only uses simple, primitive constructs. In my head `typeList` is the source of thruth, and `typeMap` is just a cache for optimization. 

I'm ok with seeing how the `LinkedMap` solution would work out when this logic is in it's own class, but maybe good to park that for a separate refactor PR until after this PR is merged?

Reply:
I don't think this is worth re-re-doing. I am ok with  the "redundant" info since it's internal and carefully kept in sync. Like I said, it's a wash for me between the visual complication of a linked list structure and the apparently simpler but somewhat redundant approach of using two distinct data structures for the "same" data. So I am marking this as resolved.

Reply:
👍